### ISSUE TITLE
Provide a sibling package for testing

### ISSUE THREAD

**Issue by snoyberg** [snoyberg/codename-karka#7]
Probably another for benchmarks. Maybe another one for console applications (optparse-applicative).

**Comment by NorfairKing** [comment]
Yes please.
Things I want in there:

```
- QuickCheck
- hspec
```

Probably also `quickcheck-instances`, but I don't actually like that package because I think `Arbitrary` is wrong.

Of course I would recommend `validity` and its sisters, but those are not a standard yet.

**Comment by snoyberg** [comment]
I'm less enthusiastic about `rio-test` than I was previously, simply because I tried to do it, and when you use the `hspec-discover` preprocessor, you have to put `hspec` in your dependency list regardless, mitigating the usefulness of this. Handling via a template that sets up a test suite + documentation may be no worse than a dedicated package, and require less maintenance.

**Comment by NorfairKing** [comment]
@snoyberg Yes, that makes sense. I'm not much-fussed about the bike-shedding that we would have to do for it. Feel free to close this issue then.
